Makefile

```diff
@@ -54,6 +55,19 @@ ui/app/script.js: $(shell find ui/app/src -iname *.elm)
 proto:
 	scripts/genproto.sh
 
+api/v2/models api/v2/restapi: $(SWAGGER) api/v2/openapi.yaml
+	-rm -r api/v2/{models,restapi}
+	cd api/v2 && swagger generate server  -f openapi.yaml --exclude-main -A alertmanager
+
+test/with_api_v2/api_v2_client/models test/with_api_v2/api_v2_client/client: $(SWAGGER) api/v2/openapi.yaml
+	-rm -r test/with_api_v1/api_v2_client; mkdir -p test/with_api_v2/api_v2_client
+	cd test/with_api_v2/api_v2_client/ && swagger generate client -f ../../../api/v2/openapi.yaml
+
+$(SWAGGER):
+	go get -u -v -d github.com/go-swagger/go-swagger/cmd/swagger
+	cd $(FIRST_GOPATH)/src/github.com/go-swagger/go-swagger; git checkout 0.15.0
+	go install github.com/go-swagger/go-swagger/cmd/swagger
```

Comment:
I would prefer it be contained within the project dir

Reply:
Fair point.

In order for it to live inside `/vendor` it would need to be imported somewhere. I see you have solved this in https://github.com/prometheus/prometheus/pull/4430 via `go generate`. This works under the prerequisite that _go-swagger_ works as a library as _vfsgen_ does. I have not yet found an easy way how to accomplish that.

_Go-swagger_ publishes a docker image with the CLI on each release. I am suggesting using that instead. As the generated API code is checked in, running docker is not a prerequisite to build Alertmanager, but only needed whenever `openapi.yaml` is touched.

@simonpasquier @stuartnelson3 what do you think?

Reply:
Yes that would work. It is also consistent with the way we generate Javascript code from Elm files so it doesn't impose more constraints on contributors (eg requiring a container runtime).